federation-js/src/composition/DirectiveMetadata.ts

```diff
@@ -168,19 +161,13 @@ export class DirectiveMetadata {
           const field = namedType.getFields()[fieldName];
           if (!field) continue;
 
-          const originalMetadata = getFederationMetadata(field);
-          let directiveUsages = originalMetadata?.directiveUsages;
-          if (directiveUsages && directiveUsages.size > 0) {
-            for (const [directiveName, usages] of directiveUsages.entries()) {
-              usages.push(...(usagesPerDirective.get(directiveName) ?? []));
```

Comment:
The bug is that this is basically pushing a thing into itself?

Reply:
So IF `directiveUsages` exists, it will only iterate over its own keys to perform the merging. New keys from the incoming "mergee" or source are currently ignored.

So in the case that this is populated with an `external` key, only `external`s will be merged and an incoming `tag` would be ignored.

Reply:
Ah I see — this was confusing enough that I didn't even understand the bug correctly!